src/core/ddsi/src/ddsi_cdrstream.c

```diff
@@ -1961,26 +2126,65 @@ static bool normalize_primarray (char * __restrict data, uint32_t * __restrict o
   return false;
 }
 
-static bool normalize_enumarray (char * __restrict data, uint32_t * __restrict off, uint32_t size, bool bswap, uint32_t num, uint32_t max) ddsrt_attribute_warn_unused_result ddsrt_nonnull_all;
-static bool normalize_enumarray (char * __restrict data, uint32_t * __restrict off, uint32_t size, bool bswap, uint32_t num, uint32_t max)
+static bool normalize_enumarray (char * __restrict data, uint32_t * __restrict off, uint32_t size, bool bswap, uint32_t flags, uint32_t num, uint32_t max) ddsrt_attribute_warn_unused_result ddsrt_nonnull_all;
+static bool normalize_enumarray (char * __restrict data, uint32_t * __restrict off, uint32_t size, bool bswap, uint32_t flags, uint32_t num, uint32_t max)
 {
-  if ((*off = check_align_prim_many (*off, size, 2, num)) == UINT32_MAX)
-    return false;
-  uint32_t * const xs = (uint32_t *) (data + *off);
-  if (bswap)
+  switch (get_enum_cdr_size (flags))
   {
-    for (uint32_t i = 0; i < num; i++)
-    {
-      xs[i] = ddsrt_bswap4u (xs[i]);
-      if (xs[i] > max)
+    case ENUM_CDR_SZ_8: {
+      if ((*off = check_align_prim_many (*off, size, 0, num)) == UINT32_MAX)
         return false;
+      uint8_t * const xs = (uint8_t *) (data + *off);
+      for (uint32_t i = 0; i < num; i++)
+        if (xs[i] > max)
+          return false;
+      *off += num;
+      break;
     }
-  }
-  else
-  {
-    for (uint32_t i = 0; i < num; i++)
-      if (xs[i] > max)
+    case ENUM_CDR_SZ_16: {
+      if ((*off = check_align_prim_many (*off, size, 1, num)) == UINT32_MAX)
+        return false;
+      uint16_t * const xs = (uint16_t *) (data + *off);
+      if (bswap)
+      {
```

Comment:
A total nitpicky suggestion: squeeze the braces a bit: I generally like it this way and I appreciate your efforts, but here the information density becomes a bit low ... No obligation but if you happen to feel the same way ...

Reply:
I've combined the swap and non-swap versions into a single line, which saves some lines of code and is still readable I think.